google/cloud/spanner_dbapi/cursor.py

```diff
@@ -394,38 +398,48 @@ def setoutputsize(self, size, column=None):
         """A no-op, raising an error if the cursor or connection is closed."""
         self._raise_if_closed()
 
+    def _handle_DQL_with_snapshot(self, snapshot, sql, params):
+        # Reference
+        #  https://googleapis.dev/python/spanner/latest/session-api.html#google.cloud.spanner_v1.session.Session.execute_sql
+        sql, params = parse_utils.sql_pyformat_args_to_spanner(sql, params)
+        res = snapshot.execute_sql(
+            sql, params=params, param_types=get_param_types(params)
+        )
+        if type(res) == int:
+            self._row_count = res
+            self._itr = None
+        else:
+            # Immediately using:
+            #   iter(response)
+            # here, because this Spanner API doesn't provide
+            # easy mechanisms to detect when only a single item
+            # is returned or many, yet mixing results that
+            # are for .fetchone() with those that would result in
+            # many items returns a RuntimeError if .fetchone() is
+            # invoked and vice versa.
+            self._result_set = res
+            # Read the first element so that the StreamedResultSet can
+            # return the metadata after a DQL statement. See issue #155.
+            while True:
+                try:
+                    self._itr = PeekIterator(self._result_set)
+                    break
+                except Aborted:
```

Comment:
A read-only snapshot will never be aborted by Cloud Spanner, so this is not necessary.

Reply:
Yeah, it also looks like something copied from another part of code (where the `execute_sql()` method is called not with `Snapshot`). Will take a look.